### PR TITLE
Feature/update net 5

### PR DESCRIPTION
**Pull request by sughosneo**
This PR contains the following changes :

- `ApplicationCore.csproj`
	- Updates project file. 
	- Updates following nuget package dependencies
		`System.Text.Json`	
- `Infrastructure.csproj`
	- Updates project file.
    - Removes old migration folders for both catalog and identity model to avoid old deprecation warnings.
    - Adds new migration plans for the net 5.0 specific update.
- `BlazorShared.csproj`
	- Updates project file.
- `PublicApi.csproj`
	- Updates project file.
	- Updates following nuget package dependencies 	
		`AutoMapper.Extensions.Microsoft.DependencyInjection`
		`MediatR`
		`MediatR.Extensions.Microsoft.DependencyInjection`
		`Swashbuckle.AspNetCore`
		`Swashbuckle.AspNetCore.SwaggerUI`
		`Swashbuckle.AspNetCore.Annotations" Version`
		`Microsoft.AspNetCore.Authentication.JwtBearer`
		`Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore`
		`Microsoft.AspNetCore.Identity.EntityFrameworkCore`
		`Microsoft.AspNetCore.Identity.UI`
		`Microsoft.EntityFrameworkCore.InMemory`
		`Microsoft.EntityFrameworkCore.SqlServer`
		`Microsoft.EntityFrameworkCore.Tools`
		`System.IdentityModel.Tokens.Jwt`
		`Microsoft.VisualStudio.Web.CodeGeneration.Design`
	- Updates Dockerfile
- `Web.csproj`
	- Updates project file. 
    - Updates `dotnet-ef` tool to latest.
	- Updates following nuget package dependencies	
		`Ardalis.Specification`
		`AutoMapper.Extensions.Microsoft.DependencyInjection`
		`MediatR`
		`MediatR.Extensions.Microsoft.DependencyInjection`
		`BuildBundlerMinifier`
		`Microsoft.AspNetCore.Components.WebAssembly.Server`
		`Microsoft.CodeAnalysis`
		`Microsoft.EntityFrameworkCore.InMemory`
		`Microsoft.VisualStudio.Web.CodeGeneration.Design`
		`Microsoft.Web.LibraryManager.Build`
		`Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore`
		`Microsoft.AspNetCore.Identity.EntityFrameworkCore`
		`Microsoft.AspNetCore.Identity.UI`
		`Microsoft.EntityFrameworkCore.SqlServer`
		`Microsoft.AspNetCore.Authentication.JwtBearer`
		`Microsoft.EntityFrameworkCore.Tools`
    - Changes `UseDatabaseErrorPage` to use `AddDatabaseDeveloperPageExceptionFilter`
- `BlazorAdmin.csproj`
	- Updates project file. 
	- Changes the project type to `Microsoft.NET.Sdk.BlazorWebAssembly`
	- Updated following nuget package dependencies		
		`Blazored.LocalStorage`
		`BlazorInputFile`
		`Microsoft.AspNetCore.Components.Authorization`
		`Microsoft.AspNetCore.Components.WebAssembly`
		`Microsoft.AspNetCore.Components.WebAssembly.Authentication`
		`Microsoft.AspNetCore.Components.WebAssembly.DevServer`
		`Microsoft.Extensions.Identity.Core`
		`Microsoft.Extensions.Logging.Configuration`
		`System.Net.Http.Json`
	- Updates `Program.cs` to load the `admin` component.
	- Updates `Create` & `Edit` Razor page for catalog items. 
	- Updates common layout of blazor theme.
	- Includes `appsettings.Development.json` for the BlazorAdmin. 
    - Changes to load admin div properly.
- `FunctionalTests.csproj`
	- Updates project file.
	- Updates following nuget package dependencies	
		`Microsoft.AspNetCore.Mvc.Testing`
		`Microsoft.NET.Test.Sdk`
		`xunit.runner.visualstudio`
		`Microsoft.EntityFrameworkCore.InMemory`
- `UnitTests.csproj`
	- Updates project file. 
	- Updates following nuget package dependencies	
		`Microsoft.NET.Test.Sdk`
		`Moq`
		`xunit.runner.visualstudio`
- `IntegrationTests.csproj`
	- Updates project file. 
	- Updates following nuget package dependencies	
		`Microsoft.EntityFrameworkCore.InMemory`
		`Microsoft.NET.Test.Sdk`
		`Moq`
		`xunit.runner.visualstudio`
- Updates `GitHub Action` workflow

### CODE REVIEW

File: src/BlazorAdmin/wwwroot/css/admin.css [hunk 1]
```diff
@@ -169,9 +168,9 @@ admin {
 }
 
 @media (min-width: 768px) {
-    admin {
+    /*admin {
```
Comment (nishanil) [inline src/BlazorAdmin/wwwroot/css/admin.css:171]: Remove commented code

File: src/PublicApi/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,11 @@
 #See https://aka.ms/containerfastmode to understand how Visual Studio uses this Dockerfile to build your images for faster debugging.
 
-FROM mcr.microsoft.com/dotnet/core/aspnet:3.1-buster-slim AS base
+FROM mcr.microsoft.com/dotnet/aspnet:5.0.0-buster-slim AS base
```
Comment (nishanil) [inline src/PublicApi/Dockerfile:3]: Use 5.0 and avoid minor versions.

**Review by nishanil — COMMENTED** [review]
Left a few comments for you to address. Can you also take a look at the GitHub Actions warnings?

File: src/PublicApi/Dockerfile (hunk 2 above)
Comment (sughosneo) [reply src/PublicApi/Dockerfile]: Removed minor versions from both the docker files.

File: src/BlazorAdmin/wwwroot/css/admin.css (hunk 1 above)
Comment (sughosneo) [reply src/BlazorAdmin/wwwroot/css/admin.css]: Removed.

File: src/PublicApi/Dockerfile (hunk 2 above)
Comment (sughosneo) [reply src/PublicApi/Dockerfile]: Removed minor versions

**Comment by sughosneo** [comment]
Thank you @nishanil for your review. Made the necessary changes as suggested.
Most of the github action specific warnings have been addressed now. But a couple of them are remaining. For e.g `Paginate(int, int)' is obsolete` in package `Ardalis.Specification`, which possibly needs details reevaluation of the existing functionlaity. 
So I have left it as of now.

**Comment by nishanil** [comment]
@ardalis Can you help with the above error please? Also, it will be great if you could review the changes.

File: src/BlazorAdmin/BlazorAdmin.csproj [hunk 3]
```diff
@@ -1,37 +1,36 @@
...
-    <PackageReference Include="Microsoft.Extensions.Logging.Configuration" Version="3.1.6" />
-    <PackageReference Include="System.Net.Http.Json" Version="3.2.0" />
-  </ItemGroup>
+	<ItemGroup>
+		<PackageReference Include="Blazored.LocalStorage" Version="3.0.0" />
+		<PackageReference Include="BlazorInputFile" Version="0.2.0" />
+		<PackageReference Include="Microsoft.AspNetCore.Components.Authorization" Version="5.0.0" />
+		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.0" />
+		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Authentication" Version="5.0.0" />
+		<!--<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Build" Version="5.0.0-preview.6.20312.15" PrivateAssets="all" />-->
```
Comment (ardalis) [inline src/BlazorAdmin/BlazorAdmin.csproj:13]: Is this needed?

**Review by ardalis — APPROVED** [review]

**Comment by ardalis** [comment]
The Specification `Paginate` functionality was shifted to use the more ubiquitous `Skip` and `Take`. You can eliminate the error by doing paging following the example shown here:
https://github.com/ardalis/Specification/blob/master/sample/Ardalis.SampleApp.Core/Specifications/CustomerSpec.cs

If you don't want to mess with it just merge and create a new issue for it and I'll take care of it.

**Comment by ardalis** [comment]
Looks good, thanks!

**Pull request merged**